highfidelity/hifi
# WL 21668 - Make sure all Prompts are correctly emitting a signal on cancel and that cancellations return null

> Window.promptAsync() and other async Window prompts should correctly emit a signal on cancel with a null value.
> 
> Run through all items marked Async to make sure they are emitting signals on cancellation and also make sure that Window.prompt() is returning null on cancel.
> 
> WindowScriptingInterface.h
> [Worklist 21668](https://worklist.net/21668)

## Test Plan
1. Copy and paste this block of code into your console:
```
    Window.promptTextChanged.connect(function(e) { print("(promptAsync) promptTextChanged: " + e); });
    Window.browseDirChanged.connect(function(e) { print("(browseDirAsync) browseDirChanged: " + e); });
    Window.browseChanged.connect(function(e) { print("(browseAsync) browseChanged: " + e); });
    Window.assetsDirChanged.connect(function(e) { print("(browseAssetsAsync) assetsDirChanged: " + e); });
    Window.saveFileChanged.connect(function(e) { print("(saveAsync) saveFileChanged: " + e); });
```

2. Run through each of the following Window methods, and confirm that closing/cancelling out of them prints the lines below in console. 

Command | Expected Result
--- | ---
Window.prompt("Hello", "world"); | `null`
Window.promptAsync("Hello", "world"); | `(promptAsync) promptTextChanged: `
Window.browseAsync() | `(browseAsync) browseChanged: `
Window.browseDirAsync("Hello world"); | `(browseDirAsync) browseDirChanged: `
Window.browseAssetsAsync("Hello world"); | `(browseAssetsAsync) assetsDirChanged: `
Window.saveAsync("Hello world"); | `(saveAsync) saveFileChanged: `
